### ISSUE TITLE
Realtime issue after 1.0.2 upgraded to 1.0.3

### ISSUE THREAD

**Issue by jpobeda** [Cacti/cacti#316]
Following issue #299 I've upgraded 1.0.2 to 1.0.3

BTW: That didn't remove the empty data collectors or input methods.

After the upgrade Realtime doesn't work anymore and I'm getting these errors on httpd.log

> [Tue Feb 21 11:26:17.163970 2017] [:error] [pid 8518] [client 143.96.129.128:59240] PHP Warning:  fopen(/usr/share/cacti-1.0.2/cache/realtime//user_njkh3p077ufaf7mhhhhu24i9u2_lgi_96.png): failed to open stream: Permission denied in /usr/share/cacti-1.0.3/lib/rrd.php on line 2036, referer: https://cacti2.xxx.yy.zz/cacti/graph_view.php?action=tree&node=tbranch-18&hgd=&hyper=true
> ERROR: opening '/usr/share/cacti-1.0.2/cache/realtime//user_njkh3p077ufaf7mhhhhu24i9u2_101.rrd': No such file or directory
> [Tue Feb 21 11:26:27.150900 2017] [:error] [pid 31079] [client 143.96.129.128:59243] PHP Warning:  fopen(/usr/share/cacti-1.0.2/cache/realtime//user_njkh3p077ufaf7mhhhhu24i9u2_lgi_96.png): failed to open stream: Permission denied in /usr/share/cacti-1.0.3/lib/rrd.php on line 2036, referer: https://cacti2.xxx.yy.zz/cacti/graph_view.php?action=tree&node=tbranch-18&hgd=&hyper=true

It seems that the upgrade process hasn't updated the "real path" of the symbolic link for cacti2 directory.

In other words

/usr/share/cacti2 was pointing to /usr/share/cacti-1.0.2
I removed that and created a new /usr/share/cacti2 pointing to /usr/share/cacti-1.0.3

Now Cacti thinks it's still the old path.

Any way/place I could fix this?

**Comment by pautiina** [comment]
Update you permission for directory
`chown -R www:www /usr/share/cacti-1.0.2/cache`
`chmod 777  /usr/share/cacti-1.0.2/cache`

**Comment by jpobeda** [comment]
File permissions are set ok.

Before to the upgrade, I removed the old sym link and added the new one, also moved the rrd files to new the path.

First bit is poiting to the old dir and second bit is poiting to the new one...

[Tue Feb 21 11:26:27.150900 2017] [:error] [pid 31079] [client 111.22.129.128:59243] PHP Warning:  # **fopen(/usr/share/cacti-1.0.2/cache/realtime//user_njkh3p077ufaf7mhhhhu24i9u2_lgi_96.png**): _**failed to open stream: Permission denied in /usr/share/cacti-1.0.3/lib/rrd.php on line 2036, referer:**_

**Comment by pautiina** [comment]
show please you permission for /usr/share/cacti-1.0.2/cache/
`ls -la /usr/share/cacti-1.0.2/cache/`
`ls -la /usr/share/cacti-1.0.2/cache/realtime/`
If you use simlink show you apache loaded module:
` httpd -t  -D DUMP_MODULES |grep alias`
I see that you cacti located on /usr/share/cacti-1.0.3 and apache try open file on /usr/share/cacti-1.0.2

**Comment by jpobeda** [comment]
alias_module (shared)
 vhost_alias_module (shared)

**Comment by pautiina** [comment]
Check your configuration Cacti path on Settings cacti. Change you directory cache to /usr/share/cacti-1.0.3/cache

**Comment by jpobeda** [comment]
I've got another installation running 1.0.1, I've just upgraded it to 1.0.3 and it works fine.
I'm gonna re-install this one anyway..give it a close or tag it unable_to_reproduce. I'll add further details if it come across the same issue.

**Comment by cigamit** [comment]
Yea, that's what I was thinking.  We will release 1.0.4 this weekend with some additional fixes.

**Comment by jpobeda** [comment]
I found the issue. My bad not checking this before but it was selinux. I'd provide a fix but it really depends on every particular setup. Since I upgraded from 1.0.2 to 1.0.3 the new dir wasn't right labeled and rrdtool wasn't able to create the files.

Basically I ran audit2allow -a -M fix and then ran semodule -i fix.cpp. This was on vm I was able to reproduce it. It's not the nicest way to fix it but..

Thanks anyway!
